clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1856,6 +1856,12 @@ class ConstantLValueEmitter : public ConstStmtVisitor<ConstantLValueEmitter,
   ConstantLValue VisitMaterializeTemporaryExpr(
                                          const MaterializeTemporaryExpr *E);
 
+  ConstantLValue emitPointerAuthSignConstant(const CallExpr *E);
+  llvm::Constant *emitPointerAuthPointer(const Expr *E);
+  unsigned emitPointerAuthKey(const Expr *E);
+  std::pair<llvm::Constant*, llvm::Constant*>
```

Comment:
From the function prototype, it's unclear that the first element of the pair stands for `StorageAddress` and the second one - for `OtherDiscriminator`. Consider adding a comment or, alternatively, using a named structure instead of `std::pair` (would break `std::tie` usage though)

Reply:
The more interesting question is the usage of Constant;  I think we can safely assume the integer component is always going to be a ConstantInt (because it's going to have to fit in a ConstantInt-shaped hole in CPA anyway.)  That hopefully makes this obvious here, and I'll propagate that in the various users separately